movement.cpp

```diff
@@ -0,0 +1,435 @@
+  *  in a clockwise motion, meaning that using a 90 degrees angle, would make the robot go to the right.
+  */
+  float pi = 3.141592;
+  int realAngle = angle + 45;
+  if (realAngle > 359)
+  {
+    realAngle -= 360;
+  }
+  float gradRadian = (realAngle * pi) / 180;
+  float pwm1 = 0, pwm2 = 0;
+  bool m1 = 0, m2 = 0;
+  float vc = 230;
+  pwm1 = vc * sin(gradRadian);
+  pwm2 = vc * cos(gradRadian);
+
+  if (pwm1 < 0)
+  {
+    m1 = true;
+    pwm1 *= -1;
+  }
+  if (pwm2 < 0)
+  {
+    m2 = true;
+    pwm2 *= -1;
+  }
+
+  if (m1)
+  {
+    front_left_motor_.changePwm(pwm1);
+    back_right_motor_.changePwm(pwm1);
+    front_left_motor_.backward();
+    back_right_motor_.backward();
+  }
+  if (!m1)
+  {
+    front_left_motor_.changePwm(pwm1);
+    back_right_motor_.changePwm(pwm1);
+    front_left_motor_.forward();
+    back_right_motor_.forward();
+  }
+  if (m2)
+  {
+    front_right_motor_.changePwm(pwm2);
+    back_left_motor_.changePwm(pwm2);
+    front_right_motor_.backward();
+    back_left_motor_.backward();
+  }
+  if (!m2)
+  {
+    front_right_motor_.changePwm(pwm2);
+    back_left_motor_.changePwm(pwm2);
+    front_right_motor_.forward();
+    back_left_motor_.forward();
+  }
+}
+
+//Function used to stop the robot.
+void Movements::stop()
+{
+  front_left_motor_.stop();
+  front_right_motor_.stop();
+  back_left_motor_.stop();
+  back_right_motor_.stop();
+}
+
+//Function used to follow the ball.
+void Movements::seguirPelota()
+{
+  /*  
+ *  -1 = Ball wasn't detected
+ *  0 = Possession of ball. This case is used to end the function.
+ *  1 = Ball in the middle of the image. In this case the robot should only advance straight forward.
+ *  2 = Ball in the top left part of the image.
+ *  3 = Ball in the top right part of the image.
+ *  4 = Ball in the bottom left part of the image.
+ *  5 = Ball in the bottom right part of the image.
+ */
+  pixy_.pixyCam(1);
+
+  int caso = -1;
+
+  if (((pixy_.pelotaX > 125) && (pixy_.pelotaX < 150)) && pixy_.pelotaY < 25)
+  {
+    caso = 0;
+  }
+  else if ((pixy_.pelotaX > 135) && (pixy_.pelotaX < 185))
+  {
+    caso = 1;
+  }
+  else if (pixy_.pelotaX > 185)
+  {
+    if (pixy_.pelotaY > 120)
+    {
+      caso = 2;
+    }
+    else if (pixy_.pelotaY < 120)
+    {
+      caso = 4;
+    }
+  }
+  else if (pixy_.pelotaX < 135)
+  {
+    if (pixy_.pelotaY > 120)
+    {
+      caso = 3;
+    }
+    else if (pixy_.pelotaY < 120)
+    {
+      caso = 5;
+    }
+  }
+
+  switch (caso)
+  {
+
+  case -1:
+    while (pixy_.pelotaX == -1)
+    {
+      int m = 0;
+      if (m <= 1)
+      {
+        movement(90);
+        delay(150);
+        stop();
+        delay(50);
+        pixy_.pixyCam(1);
+        m++;
+      }
+      else
+      {
+        movement(270);
+        delay(300);
+        stop();
+        delay(50);
+        pixy_.pixyCam(1);
+        m = 0;
+      }
+    }
+    break;
+
+  case 0:
+    break;
+
+  case 1:
+    movement(0);
+    delay(25);
+    stop();
+    delay(5);
```

Comment:
Why all the delays?  It should be dynamic, meaning it should change whenever it needs to.